src/cljs/district_voting/pages/vote_page.cljs

```diff
@@ -4,27 +4,47 @@
     [district-voting.components.contract-info :refer [contract-info]]
     [district-voting.components.how-to-instructions :refer [how-to-instructions]]
     [district-voting.components.voting-bar :refer [voting-bar]]
-    [district-voting.constants :as constants]
     [district-voting.styles :as styles]
     [district0x.components.misc :as misc :refer [row row-with-cols col center-layout paper page]]
-    [re-frame.core :refer [subscribe dispatch]]))
+    [district-voting.proposals.subs :as proposal-subs]
+    [markdown.core :refer [md->html]]
+    [re-frame.core :refer [subscribe dispatch]]
+    [reagent.core :as r]
+    [cljs-react-material-ui.reagent :as ui]
+    [district-voting.components.expandable-text :refer [expandable-text]]
+    )
+  (:require-macros [reagent.ratom :refer [reaction]]))
 
 (defn link [name url]
   [:a {:href url
        :target :_blank}
    name])
 
+(def sort-options {:votes "DNT Votes"
+                   :gh-upvotes "Github upvotes"
+                   :gh-comments "Github comments"
+                   :latest "Latest"})
+
 (defmethod page :route.vote/home []
   (let [votes (subscribe [:voting/candidates-voters-dnt-total :next-district])
         votes-total (subscribe [:voting/voters-dnt-total :next-district])
         loading? (subscribe [:voting-loading? :next-district])
         can-submit? (subscribe [:district0x/can-submit-into-blockchain?])
-        vote-form (subscribe [:form.next-district/vote])]
+        vote-form (subscribe [:form.next-district/vote])
+        proposals (subscribe [::proposal-subs/list :next-district])
+        limit (r/atom 10)
+        sort-order (r/atom :votes)
```

Comment:
This question is an important point of app's architecture IMO. Yes, these are certainly pros. But there are cons: storing component-related state globally, is a leaked incapsulation, creates intercomponent entanglement, which becomes real PITA with app's growth, creating such abominations as registering components ID's in db, making app's state almost unreadable. I wholehartedly agree with reagent's approach on isolates state, and i use a litmus question for self-guidance "Is this data a source of truth for an application?" If yes, it goes to global state. If it derivates from another data in the state, then it should be in some reaction. If it's relatives to the components, and it's state is not valuable enough to be persisted between session, then it's component's state, you can refactor them them out the libraries and have multiple instances (like in-place editors) whithout interfering with app's db schema. Like "muscle memory" frees up brain for important things.